src/calculator/index.ts

```diff
@@ -297,48 +300,63 @@ export default class Calculator {
       `${this.chainId}/rounds.json`
     );
 
-    const round = rounds.find((round) => round.id === this.roundId) as Round;
+    const round = rounds.find((round) => round.id === this.roundId);
+
+    if (round === undefined) {
+      throw new ResourceNotFoundError("round");
+    }
 
     const currentResults = await this._calculate(votes);
-    const conversionRates = await Promise.all(
-      potentialVotes.map(async (vote) => [
-        vote.token,
-        await this.priceProvider.getUSDConversionRate(this.chainId, vote.token),
-      ])
-    );
-    const prices = Object.fromEntries(conversionRates) as Record<
-      string,
-      Price & { decimals: number }
-    >;
+    const prices: Record<string, PriceWithDecimals> = {};
+
+    // fetch each token price only once
+    for (const vote of potentialVotes) {
+      if (prices[vote.token] === undefined) {
+        prices[vote.token] = await this.priceProvider.getUSDConversionRate(
+          this.chainId,
+          vote.token
+        );
+      }
+    }
+
     const conversionRateRoundToken =
       await this.priceProvider.getUSDConversionRate(this.chainId, round.token);
 
-    const potentialVotesAugmented: Vote[] = await Promise.all(
-      potentialVotes.map((vote) => {
-        const tokenPrice = prices[vote.token];
-        const amountUSD =
-          Number(formatUnits(vote.amount, 18)) * tokenPrice.price;
-
-        /*TODO: take into account round matching token decimals instead of default 18 */
-        const amountRoundToken =
-          (Number(amountUSD) / conversionRateRoundToken.price) *
-          Math.pow(10, 18);
-
-        return {
-          ...vote,
-          amount: vote.amount.toString(),
-          amountRoundToken: Math.floor(amountRoundToken).toString(),
-          amountUSD,
-          applicationId: vote.applicationId,
-          id: "",
-        };
-      })
-    );
+    const potentialVotesAugmented: Vote[] = potentialVotes.map((vote) => {
+      const tokenPrice = prices[vote.token];
 
-    const potentialResults = await this._calculate([
-      ...votes,
-      ...potentialVotesAugmented,
-    ]);
+      const amountUSD = convertTokenToFiat({
+        tokenAmount: vote.amount,
+        tokenDecimals: tokenPrice.decimals,
+        price: tokenPrice.price,
+        priceDecimals: 10,
+      });
+
+      const amountRoundToken = convertFiatToToken({
```

Comment:
Is `round` here is meant as in `Math.round` rather than "funding round"? If so, I'd suggest changing it to something able to remove all doubts, like `roundedTokenAmount` (key point being `roundED`).